### PR TITLE
Add docs “Translating WooCommerce”

### PR DESCRIPTION
**Pull request by nielslange**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Add 'Translating WooCommerce' documentation to guide external contributors on translation contributions.

Closes https://github.com/woocommerce/woodocs/issues/32.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Look up the added docs file.
2. Verify that all links are working.
3. Verify that all screenshots are matching.
4. Verify that the instructions are correct.

### Changelog entry

n/a

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: docs/localization-translation/README.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+To set your WordPress site's language:
+
+1. Go to `WP Admin » Settings » General` and adjust the `Site Language`.
+2. Go to `WP Admin » Dashboard » Updates` and click the `Update Translations` button.
+
+Once this has been done, the shop displays in your locale if the language file exists. Otherwise, you need to create the language files (process explained below).
+
+## Contributing your localization to core
+
+We encourage contributions to our translations. If you want to add translated strings or start a new translation, simply register at WordPress.org and submit your translations to [translate.wordpress.org](https://translate.wordpress.org/projects/wp-plugins/woocommerce/) for approval.
```
Comment (nerrad) [inline docs/localization-translation/README.md]: The mismatch between the URL stub (translate.wordpress.org) and the full path it links to can be misleading - especially given the perceived semantic meaning of the visible URL "translate...WordPress". While less apparent when viewing the markdown, the rendered view had me thinking this wasn't going to link to the right place.

When URL is a label, I think it'd be better to either include the full path as the label, or link "submit your translations for approval" as the linked label. For example:

```md
...and [submit your translations for approval](https://translate.wordpress.org/projects/wp-plugins/woocommerce/).
```

_This comment applies to elsewhere in the doc where the similar pattern is found._

File: docs/localization-translation/README.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+3. Click the heading under `Set/Sub Project` to view and download a Stable version.
+
+    ![screenshot](https://woocommerce.com/wp-content/uploads/2012/01/2016-02-17-at-09.59.png)
+
+4. Scroll to the bottom for export options. Export a `.mo` file for use on your site.
+
+    ![screenshot](https://woocommerce.com/wp-content/uploads/2012/01/2016-02-17-at-10.00.png)
+
+5. Rename this file to `woocommerce-YOURLANG.mo` (e.g., Great Britain English should be `en_GB`).
```
Comment (nerrad) [inline docs/localization-translation/README.md]: Would be good to link to the acceptable list of language codes.

File: docs/localization-translation/README.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+6. Save when finished.
+
+    ![screenshot](https://woocommerce.com/wp-content/uploads/2012/01/loco-translate-translating-1.png)
+
+    > **Note:** When a new version of WooCommerce is released and updated on your site, you need refresh the .po file to look for new strings. Use the “Sync” button to find any new string to start translating.
+
+## Creating custom translations with PoEdit
+
+[Poedit ](https://poedit.net/) is a more advanced alternative to the [Loco Translate](https://wordpress.org/plugins/loco-translate/) plugin. Only advanced users or translators wishing to create a custom translation to WooCommerce should attempt this method.
```
Comment (nerrad) [inline docs/localization-translation/README.md]: > Only advanced users or translators wishing to create a custom translation to WooCommerce should attempt this method.

I would drop this superlative sentence for a couple reasons:

- Poedit has already been referred to as an advanced alternative.
- Loco Translate has already been referred to as a way to create a custom translation (using Poedit is not distinctive on that point).

**Review by nerrad — COMMENTED** [review]
Thanks Niels, I've left a few inline comments. Also:

- Referencing other third-party tools introduces the risk of documentation going stale (changing interfaces referenced in screenshots etc). To avoid this, I recommend just linking to these third-party tool docs in an "Other tools" section and then add a caveat along the lines of "There are some other third-party tools that can help with translations". The exception for this comment would be to keep the documentation on PoEdit as it's been around for a long time and it's less risky to include it.
- There is no mention of JS translations in this doc. I'm assuming this is something you're aware of but thought I'd mention it just in case. I think it's fine to migrate this as an initial iteration, however, it'd be good to create a follow-up issue to improve it with some references to JS translations (especially given using Poedit won't cover those strings).

File: docs/localization-translation/README.md (hunk 1 above)
Comment (nielslange) [reply docs/localization-translation/README.md]: Thanks for pointing this out. As we're showing the URLs in different contexts, I decided to show the full URL in all cases.

File: docs/localization-translation/README.md (hunk 2 above)
Comment (nielslange) [reply docs/localization-translation/README.md]: I've added steps that explain how to locate the language code directly on the language pages, e.g. on https://translate.wordpress.org/locale/en-gb/default/wp-plugins/woocommerce/, and added the following screenshot:

Screenshot 2023-10-17 at 09 44 53

File: docs/localization-translation/README.md (hunk 3 above)
Comment (nielslange) [reply docs/localization-translation/README.md]: I've removed the superlative sentence.

**Comment by nielslange** [comment]
Thanks for your review, @nerrad. 🙌

> Referencing other third-party tools introduces the risk of documentation going stale (changing interfaces referenced in screenshots etc). To avoid this, I recommend just linking to these third-party tool docs in an "Other tools" section and then add a caveat along the lines of "There are some other third-party tools that can help with translations". The exception for this comment would be to keep the documentation on PoEdit as it's been around for a long time and it's less risky to include it.

I see what you mean regarding the third-party tools. I've removed the Loco Translate section and introduced the suggested "Other tools" section where I've mentioned the tool.

> There is no mention of JS translations in this doc. I'm assuming this is something you're aware of but thought I'd mention it just in case. I think it's fine to migrate this as an initial iteration, however, it'd be good to create a follow-up issue to improve it with some references to JS translations (especially given using Poedit won't cover those strings).

Regarding JS translations, you're right. While I'm aware of its importance, we do have extensive documentation on handling JS translations in WooCommerce Blocks, e.g. [Translations in JS/TS files](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/translations/translations-in-JS-TS-files.md#translations-in-jsts-files) and [Loading translations for TJS/TS files](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/translations/translation-loading.md#loading-translations-for-tjsts-files). My plan is to incorporate this information once WooCommerce Blocks has been merged into core. In this first version, I focused more on what @brentmackinnon suggested regarding the translation guidelines in pdzGYk-1l3-p2.

**Review by nerrad — APPROVED** [review]
Thanks for addressing the feedback. This is a good i1.

**Comment by nielslange** [comment]
Thanks again for reviewing it, @nerrad.

**Pull request merged**